src/main/java/it/gov/pagopa/pu/workflow/wf/receiptclassification/iuf/classification/IufReceiptClassificationWFImpl.java

```diff
@@ -0,0 +1,122 @@
+
+    clearClassifyIufActivity = wfConfig.buildClearClassifyIufActivityStub();
+    iufClassificationActivity = wfConfig.buildIufClassificationActivityStub();
+    transferClassificationActivity = wfConfig.buildTransferClassificationActivityStub();
+
+    transferClassificationWfHelperActivity = wfConfig.buildTransferClassificationStarterHelperActivityStub();
+
+  }
+
+  @Override
+  public void classify() {
+
+    if (result != null && result.isSuccess()) {
+
+      // signal transfer classification for each transfer
+      result.getTransfers2classify().forEach(transfer2ClassifyDTO -> {
+
+        String iuv = transfer2ClassifyDTO.getIuv();
+        String iur = transfer2ClassifyDTO.getIur();
+        int transferIndex = transfer2ClassifyDTO.getTransferIndex();
+
+        transferClassificationWfHelperActivity.signalTransferClassificationWithStart(result.getOrganizationId(), iuv, iur, transferIndex);
+      });
+
+    } else {
+      log.warn("Result is null or classification was not successful for organizationId {}",
+        result != null ? result.getOrganizationId() : "N/A");
+    }
+
+  }
+
+
+  @Override
+  public void signalForTreasury(IufReceiptClassificationForTreasurySignalDTO signalDTO) {
+
+    log.info("Handling iuf receipt classification for treasury with organizationId {}, treasuryId {} and IUF {}",
+      signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+    boolean clearedForTreasury = clearClassifyIufActivity.deleteClassificationByIuf(
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf());
+
+    log.info("IUF receipt classification cleared with result {} for organizationId {}, treasuryId {} and IUF {}",
+      clearedForTreasury,
+      signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+    result = iufClassificationActivity.classify(signalDTO.getOrganizationId(), signalDTO.getTreasuryId(), signalDTO.getIuf());
+
+  }
+
+  @Override
+  public void signalForReporting(IufReceiptClassificationForReportingSignalDTO signalDTO) {
+
+    log.info("Handling iuf receipt classification for reporting with organizationId {}, IUF {} and outcome {}",
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf(),
+      signalDTO.getOutcomeCode()
+    );
+
+    boolean clearedForTreasury = clearClassifyIufActivity.deleteClassificationByIuf(
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf());
+
+    log.info("IUF receipt classification for reporting cleared. Result is {} for organizationId {}, iuf {} and outcome {}",
+      clearedForTreasury,
+      signalDTO.getOrganizationId(),
+      signalDTO.getIuf(),
+      signalDTO.getOutcomeCode());
+
+    result = IufClassificationActivityResult.builder()
+      .organizationId(signalDTO.getOrganizationId())
+      .success(true)
+      .transfers2classify(signalDTO.getTransfers2classify())
+      .build();
```

Comment:
this should add the new IufClassificationActivityResult to result instance variable